src/compiler/transformers/classFields.ts

```diff
@@ -137,14 +179,217 @@ namespace ts {
+            left: PrivateIdentifierPropertyAccessExpression;
+        }
+
+        function isPrivateIdentifierAssignmentExpression(node: Node): node is PrivateIdentifierAssignmentExpression {
+            return isAssignmentExpression(node) && isPrivateIdentifierPropertyAccessExpression(node.left);
+        }
+
+        function visitBinaryExpression(node: BinaryExpression) {
+            if (isDestructuringAssignment(node)) {
+                const left = transformDestructuringAssignmentTarget(node.left);
+                if (left !== node.left) {
+                    return updateBinary(node, left, visitNode(node.right, visitor), node.operatorToken);
+                }
+            }
+            else if (isPrivateIdentifierAssignmentExpression(node)) {
+                const privateIdentifierInfo = accessPrivateIdentifier(node.left.name);
+                if (privateIdentifierInfo) {
+                    switch (privateIdentifierInfo.placement) {
+                        case PrivateIdentifierPlacement.InstanceField: {
+                            return transformPrivateIdentifierInstanceFieldAssignment(privateIdentifierInfo, node);
+                        }
+                    }
+                }
             }
+            return visitEachChild(node, visitor, context);
+        }
 
+        function transformPrivateIdentifierInstanceFieldAssignment(privateIdentifierInfo: PrivateIdentifierInstanceField, node: PrivateIdentifierAssignmentExpression) {
+            if (isCompoundAssignment(node.operatorToken.kind)) {
+                const isReceiverInlineable = isSimpleInlineableExpression(node.left.expression);
+                const getReceiver = isReceiverInlineable ? node.left.expression : createTempVariable(hoistVariableDeclaration);
+                const setReceiver = isReceiverInlineable
+                    ? node.left.expression
+                    : createAssignment(getReceiver, node.left.expression);
+                return setOriginalNode(
+                    createClassPrivateFieldSetHelper(
+                        context,
+                        setReceiver,
+                        privateIdentifierInfo.weakMapName,
+                        createBinary(
+                            createClassPrivateFieldGetHelper(
+                                context,
+                                getReceiver,
+                                privateIdentifierInfo.weakMapName
+                            ),
+                            getOperatorForCompoundAssignment(node.operatorToken.kind),
+                            visitNode(node.right, visitor)
+                        )
+                    ),
+                    node
+                );
+            }
+            else {
+                return setOriginalNode(
+                    createClassPrivateFieldSetHelper(
+                        context,
+                        node.left.expression,

```

Comment:
You are not visiting `node.left.expression` here. You might want to add a test case for something like:

```ts
obj[(new class { #x = 1; readonly s = "prop" }).s].#y = 1;
```